src/core/types/object/json.js

```diff
@@ -111,7 +111,11 @@ export function deepEquals(a, b, depth = 5) {
           return false;
         }
         for (const k of keysA) {
-          queue.push({a: a[k], b: b[k], depth: depth - 1});
+          queue.push({
+            a: /** @type {any} */ (a)[k],
+            b: /** @type {any} */ (b)[k],
```

Comment:
```suggestion
            a: /** @type {*} */ (a)[k],
            b: /** @type {*} */ (b)[k],
```

Reply:
CC is `*`, TS is `any`. I'd rather not commit this suggestion unless you feel strongly in which equivalent syntax is used.

Reply:
Saw your earlier comment now, updating.